Example in build system section fails

I am trying to go through the example in the 'Build system' section of the docs, but when I do `dune build @melange` it fails with:

```
File "dune-project", line 1, characters 11-14:
1 | (lang dune 3.8)
               ^^^
Error: Version 3.8 of the dune language is not supported.
Supported versions of this extension in version 3.8 of the dune language:
- 1.0 to 1.12
- 2.0 to 2.9
- 3.0 to 3.7
```

(I'm on MacOS (ARM processor))

`opam pin add dune.dev --dev-repo` worked but `opam pin add melange.dev --dev-repo` failed with:

```
[ERROR] The compilation of melange.dev failed at "dune build -p melange -j 7 @install".

#=== ERROR while compiling melange.dev ========================================#
# context     2.1.4 | macos/arm64 | ocaml-base-compiler.4.14.1 | pinned(git+https://github.com/melange-re/melange.git#ff0ca826)
# path        ~/Learning/Melange/_opam/.opam-switch/build/melange.dev
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p melange -j 7 @install
# exit-code   1
# env-file    ~/.opam/log/melange-24423-e5b505.env
# output-file ~/.opam/log/melange-24423-e5b505.out
### output ###
# File "vendor/dune", line 11, characters 35-50:
# 11 |    (modules_without_implementation melange_wrapper)))
#                                         ^^^^^^^^^^^^^^^
# Error: Module Melange_wrapper doesn't exist.
```

(It did hang the first time I tried it so I had to `ctrl-c` out of it and do it again.)

--- a/docs/build-system.md
+++ b/docs/build-system.md
@@ -71,8 +71,17 @@ opam switch create . 4.14.1 --deps-only
 Install Dune and Melange in the switch:
 
 ```bash
-opam install dune melange
+opam pin add dune.dev --dev-repo
+opam pin add melange.dev --dev-repo
 ```
+
+<div class="text-reasonml">
+As we will be using Reason syntax, let’s install the `reason` package too:
+
+```bash
+opam install reason
+```
+</div>
 
 Create a file named `dune-project`. This file will tell Dune a few things about
 our project configuration:

